### PR TITLE
DHP-93 confirm fitbit registration

### PR DESCRIPTION
**Pull request by yonidoronpeters**
## Description
This PR adds UI components for showing Connected Devices and Devices to Connect sections. 
It also shows a success alert when fitbit account connections are successful or unsuccessful.

## Testing done
Unit tests for components

## Screenshots
Screen Shot 2022-04-13 at 2 29 17 PM
Screen Shot 2022-04-13 at 2 29 07 PM

## Acceptance criteria
- [x] Given the Fitbit registration is successful
When I am navigated back to VA.gov
Then I am served a notification that confirms the device registration
- [x] Given the Fitbit registration is unsuccessful
When I am navigated back to VA.gov
Then I am served a notification telling me the device registration failed

## Definition of done
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const DeviceDisconnectionCard = ({ vendor, onClickHandler }) => {
+  return (
+    <div className="connect-device">
+      <h3 className="vads-u-margin-y--0">{vendor}</h3>
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:7]: I'm looking at the wire frames, and it looks like we're just missing `- Connected` after the vendor name. So, in this section it should look like:
**Vendor** - Connected
(button) Disconnect

I also just noticed a small difference between the `Disconnect` button and the wireframe. The wireframe has the background of the button white, whereas it looks grey here.

Comment (TravisGM) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:7]: @yonidoronpeters

**Review by TravisGM — CHANGES_REQUESTED** [review]

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx (hunk 1 above)
Comment (yonidoronpeters) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:7]: Good catch! will update to "Fitbit - Connected"

The button we use has a clear background. I don't see a button with a white background in [Storybook](https://design.va.gov/storybook/?path=/docs/components-buttons--primary#secondary-buttons) so wonder if the white background was by design or by mistake. @toddstanich thoughts on the button?

Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:7]: Pushing the `- Connected` updated as well as SASS for a button with a white background. It's commented out though pending @toddstanich's thoughts.

Comment (reblg) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx]: I also noticed a discrepancy in the page title which I updated.

Comment (toddstanich) [reply src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx]: We can leave the disconnect button as is. It looks like it is an error in the Sketch design library and the component does not have a white BG.

File: src/applications/dhp-connected-devices/containers/App.jsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ export default function App() {
       <div className="usa-width-three-fourths">
         <article className="usa-content">
           <div className="schemaform-title">
-            <h1>Connect your health devices</h1>
+            <h1>Connect your health devices to share data</h1>
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/containers/App.jsx:14]: Great catch here. Unfortunately, I also caught something in the process. Not sure if we want to be this nit picky, but the wire frames do show that it should say `Connect your health device to share data`. Personally, I think what you all have here makes more sense than the wire frames. Pluralizing it seems more grammatically correct.

File: src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export const DeviceConnectionSucceededAlert = () => {
+  return (
+    <div data-testId="success-alert">
+      <va-alert
+        close-btn-aria-label="Close notification"
+        status="success"
+        visible
+      >
+        <h3 slot="headline">Device Connected</h3>
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx:11]: Totally looks fine to me, but the wire frames do show this as `Device connected`, not `Device Connected`.

File: src/applications/dhp-connected-devices/components/DeviceConnectionCard.jsx [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
-      </p>
-      <p>
-        <a href={authUrl}>Connect</a>
+      <h3 className="vads-u-margin-y--0">{device.vendor}</h3>
+      <p className="vads-u-margin-y--0">
+        <a
+          data-testId={`${device.vendor}-connect-link`}
+          id={`${device.vendor}-connect-link`}
+          className="connect-button"
+          href={`${environment.API_URL}/dhp_connected_devices${device.authUrl}`}
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DeviceConnectionCard.jsx:14]: I see the change here, nice work!

File: src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <div data-testId="failure-alert">
+      <va-alert
+        close-btn-aria-label="Close notification"
+        status="error"
+        visible
+      >
+        <h3 slot="headline">We couldn’t connect your device</h3>
+        <div>
+          We were not able to connect to your device now. Please try again.
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx:28]: Same thing here. The wire frames show `We were not able to connect to your device _right_ now`. Totally looks good to me, just an FYI.

File: src/applications/dhp-connected-devices/components/DevicesToConnectSection.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      }
+      return <></>;
+    });
+  };
+
+  return (
+    <>
+      {areAllDevicesConnected() && (
+        <p data-testId="all-devices-connected-alert">
+          You have connected all supported devices
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DevicesToConnectSection.jsx:22]: I may have an outdated reference for wire frames, but what I see on mine is different than here.

I see some kind of H tag; "Devices you can connect" and underneath it:
"There are no devices able to connect"

**Review by TravisGM — CHANGES_REQUESTED** [review]

File: src/applications/dhp-connected-devices/containers/App.jsx (hunk 2 above)
Comment (toddstanich) [reply src/applications/dhp-connected-devices/containers/App.jsx:14]: Agree that what you have with "devices" makes more sense. I'll update the designs to match.

File: src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx (hunk 3 above)
Comment (toddstanich) [reply src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx:11]: Let's use "Device connected" In general, VA uses sentence case.

File: src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx (hunk 5 above)
Comment (toddstanich) [reply src/applications/dhp-connected-devices/components/DeviceConnectionAlerts.jsx:28]: Good catch, let's update to "We were not able to connect to your device right now. Please try again."

File: src/applications/dhp-connected-devices/components/DevicesToConnectSection.jsx (hunk 6 above)
Comment (toddstanich) [reply src/applications/dhp-connected-devices/components/DevicesToConnectSection.jsx:22]: Let's use, "There are no devices available to connect."

File: src/applications/dhp-connected-devices/containers/App.jsx (hunk 2 above)
Comment (reblg) [reply src/applications/dhp-connected-devices/containers/App.jsx:14]: Thanks!

File: src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const DeviceDisconnectionCard = ({ device, onClickHandler }) => {
+  return (
+    <div className="connect-device">
+      <h3 className="vads-u-margin-y--0">
+        {device.vendor}{' '}
+        <span className="connected-header-text"> - Connected</span>{' '}
```
Comment (TravisGM) [inline src/applications/dhp-connected-devices/components/DeviceDisconnectionCard.jsx:9]: 😊

**Review by TravisGM — APPROVED** [review]
Looks good to me! Good work team! 💃

**Pull request merged**